File: benchmarks/bench_hsi.py
import xarray as xr, numpy as np, pandas as pd

print("🟣 [PYTHON] HSI: Correlation Matrix")

# 1. LOAD
ds = xr.open_zarr("data/pavia.zarr")
# Shape: (X, Y, Band). We need (Pixels, Band)
# This reshape forces a massive memory shuffle
flat = ds['reflectance'].stack(pixel=("x", "y"))

# 2. CORRELATION MATRIX (Heavy Linear Algebra)
# Corrcoef on (Band, Pixel) -> Returns (Band, Band) matrix
# .data converts to numpy (triggering Dask compute)
matrix = np.corrcoef(flat.data)

# 3. SAVE
pd.DataFrame(matrix).to_csv("results/py_corr_matrix.csv")
print("✅ Complete")

File: benchmarks/bench_hsi.jl
using YAXArrays, Zarr, Statistics, CSV, DataFrames

println("🟣 [JULIA] HSI: Correlation Matrix")

# 1. LOAD
cube = Cube("data/pavia.zarr")

# 2. RESHAPE (Lazy)
# YAXArrays doesn't support easy stack(), so we read to memory for the Algebra stress
full_data = read(cube) # Reads into RAM
s = size(full_data)
flat = reshape(full_data, s[1]*s[2], s[3])

# 3. CORRELATION (Linear Algebra)
corr_mat = cor(flat)

# 4. SAVE
CSV.write("results/jl_corr_matrix.csv", Tables.table(corr_mat))
println("✅ Complete")
